common-lib/src/main/scala/com/gu/mediaservice/model/MimeType.scala

```diff
@@ -0,0 +1,53 @@
+package com.gu.mediaservice.model
+
+import play.api.Logger
+import play.api.libs.json._
+
+class UnsupportedMimeTypeException(val mimeType: String) extends Exception
+
+sealed trait MimeType {
+  def name: String = this match {
+    case Jpeg => "image/jpeg"
+    case Png => "image/png"
+    case Tiff => "image/tiff"
+  }
+
+  def fileExtension: String = s".${name.split('/').reverse.head}"
+
+  override def toString: String = this.name
+}
+
+object MimeType {
+  def apply(value: String): MimeType = value.toLowerCase match {
+    case "image/jpeg" => Jpeg
+    case "image/png" => Png
+    case "image/tiff" => Tiff
+
+    // Support crops created in the early years of Grid (~2016) which state mime type w/out an 'image/' prefix
+    // TODO correct these values in a reindex
+    case "jpg" => {
+      Logger.info("Encountered legacy mime type representation")
```

Comment:
Is it useful to log out the type here as well? (It may well be not useful at all, I'm just wondering if we'd like to know numbers etc.)

Reply:
Added. I've not made it a marker or anything as we should be able to run a bulk update query on ES to correct things.